### PR TITLE
EmberApp: Use "src/ui/index.html" if it exists

### LINKED ISSUE #6818: Allow `index.html` to be in `src/ui/` folder
For the implementation of the [Module Unification RFC](https://github.com/emberjs/rfcs/pull/143) we need to change the `EmberApp` class to allow the `index.html` file to be placed in the `src/ui/` folder instead of `app/`.

see also https://github.com/emberjs/ember.js/issues/14882

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #6818 

/cc @mixonic

### CODE REVIEW

File: lib/broccoli/ember-app.js [hunk 1]
```diff
@@ -645,20 +649,16 @@ class EmberApp {
...
-          relativePath = htmlName;
-        }
-        return relativePath;
-      },
-      annotation: 'Funnel: index.html',
-    });
+    let index;
+    if (!this.trees.src) {
+      index = this._rawAppIndex(htmlName);
+
```
Comment (rwjblue) [inline lib/broccoli/ember-app.js]: nit-pick: extra whitespace line here

Comment (Turbo87) [inline lib/broccoli/ember-app.js]: yeah, that's somewhat deliberate, but I can remove it if you're feeling annoyed by it ;)

**Comment by rwjblue** [comment]
@homu r+

**Comment by Turbo87** [comment]
@homu retry

**Comment by mixonic** [comment]
great thanks @Turbo87 :-D

**Comment by iezer** [comment]
Hi @Turbo87 Do I need to set a feature flag to test this behavior? If so, how do I do that? I tried to test this by creating a new app using `ember-cli#master`, moving `app/index.html` and `app/styles` to `src/ui`, and booting up the app. Moving the styles worked fine but moving `index.html` causes build errors that ember-cli cannot find `index.html`.

Can you please give this shot and let me know if it works for you? Also I'm not sure how to set the `MODULE_UNIFICATION` feature flag for `ember-cli` so if I need to set that, that's probably the first step. Thanks.

I did the following steps.
 - `npm install -g ember-cli/ember-cli#master`  install ember-cli master globally
 - `ember new module-test` create a new app
 - Update new app's `package.json` to get ember-cli from `ember-cli/ember-cli#master`
 - `mv app/index.html src/ui/`
 - `mv app/styles src/ui/`

**Pull request merged**